Chapter1/4-Shaders/Window.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using OpenTK;
+using OpenTK.Graphics;
+using OpenTK.Graphics.OpenGL4;
+using OpenTK.Input;
+using LearnOpenTK.Common;
+
+namespace LearnOpenTK
+
+{   // Here we'll be elaborating on what shaders can do from the Hello World project we worked on before
+    // Specifically we'll be showing how shaders deal with input and output from the main program 
+    // And between each other
+    public class Window : GameWindow
+    {
+
+        private readonly float[] _vertices =
+        {
+            -0.5f, -0.5f, 0.0f, // Bottom-left vertex
+             0.5f, -0.5f, 0.0f, // Bottom-right vertex
+             0.0f,  0.5f, 0.0f  // Top vertex
+        };
+
+        private int _vertexBufferObject;
+
+        private int _vertexArrayObject;
+
+        private Shader _shader;
+
+        public Window(int width, int height, string title)
+            : base(width, height, GraphicsMode.Default, title)
+        {
+        }
+
+        protected override void OnLoad(EventArgs e)
+        {
+            GL.ClearColor(0.2f, 0.3f, 0.3f, 1.0f);
+
+            _vertexBufferObject = GL.GenBuffer();
+
+            GL.BindBuffer(BufferTarget.ArrayBuffer, _vertexBufferObject);
+            GL.BufferData(BufferTarget.ArrayBuffer, _vertices.Length * sizeof(float), _vertices, BufferUsageHint.StaticDraw);
+
+            _shader = new Shader("Shaders/shader.vert", "Shaders/shader.frag");
+
+            _shader.Use();
+
+            _vertexArrayObject = GL.GenVertexArray();
+            GL.BindVertexArray(_vertexArrayObject);
+
+            GL.VertexAttribPointer(0, 3, VertexAttribPointerType.Float, false, 3 * sizeof(float), 0);
+            GL.EnableVertexAttribArray(0);
+
+            GL.BindBuffer(BufferTarget.ArrayBuffer, _vertexBufferObject);
+
+            // Vertex attributes are the we send input into the vertex shader from the main program 
+            // So here we're checking to see how many vertex attributes our hardware can handle
+            // OpenGL at minimum supports 16 vertex attributes
```

Comment:
idk if its necesary, but maybe explain that this is just an optional debugging/logging step? idk.